src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -268,6 +275,10 @@ protected virtual void OnConnectorAdded(ConnectorModel obj)
             RegisterConnector(obj);
             var handler = ConnectorAdded;
             if (handler != null) handler(obj);
+            if (workspaceLoaded)
```

Comment:
For file load, update the upstream collection only after the workspace is added.